tests/foreman/ui_airgun/test_jobinvocation.py

```diff
@@ -0,0 +1,174 @@
+        bridge=settings.vlan_networking.bridge,
+    ) as client:
+        client.install_katello_ca()
+        client.register_contenthost(module_org.label, lce='Library')
+        assert client.subscribed
+        add_remote_execution_ssh_key(client.ip_addr)
+        Host.update({
+            'name': client.hostname,
+            'subnet-id': module_subnet.id,
+        })
+        # connect to host by ip
+        Host.set_parameter({
+            'host': client.hostname,
+            'name': 'remote_execution_connect_by_ip',
+            'value': 'True',
+        })
+        hostname = client.hostname
+        with session:
+            session.organization.select(module_org.name)
+            assert session.host.search(hostname)[0]['Name'] == hostname
+            session.jobinvocation.run({
+                'job_category': 'Commands',
+                'job_template': 'Run Command - SSH Default',
+                'search_query': 'name ^ {}'.format(hostname),
+                'template_content.command': 'ls',
+            })
+            session.jobinvocation.wait_job_invocation_state(
+                entity_name='Run ls', host_name=hostname)
+            status = session.jobinvocation.read(
+                entity_name='Run ls', host_name=hostname)
+            assert status['overview']['hosts_table'][0]['Status'] == 'success'
+
+
+@tier3
+def test_positive_run_custom_job_template_by_ip(
+        session, module_org, module_subnet):
+    """Run a job template on a host connected by ip
+
+    :id: e283ae09-8b14-4ce1-9a76-c1bbd511d58c
+
+    :Setup: Create a working job template.
+
+    :Steps:
+
+        1. Set remote_execution_connect_by_ip on host to true
+        2. Navigate to an individual host and click Run Job
+        3. Select the job and appropriate template
+        4. Run the job
+
+    :expectedresults: Verify the job was successfully ran against the host
+
+    :CaseLevel: System
+    """
+    with VirtualMachine(
+            distro=DISTRO_RHEL6,
+            provisioning_server=settings.compute_resources.libvirt_hostname,
+            bridge=settings.vlan_networking.bridge,
+    ) as client:
+        client.install_katello_ca()
+        client.register_contenthost(module_org.label, lce='Library')
+        assert client.subscribed
+        add_remote_execution_ssh_key(client.ip_addr)
+        Host.update({
+            'name': client.hostname,
+            'subnet-id': module_subnet.id,
+        })
+        # connect to host by ip
+        Host.set_parameter({
+            'host': client.hostname,
+            'name': 'remote_execution_connect_by_ip',
+            'value': 'True',
+        })
+        hostname = client.hostname
+        job_template_name = gen_string('alpha')
+        with session:
+            session.organization.select(module_org.name)
+            assert session.host.search(hostname)[0]['Name'] == hostname
+            session.jobtemplate.create({
```

Comment:
wouldn't it be better to create this using nailgun?

Reply:
You are right, but I want to have at least one end-to-end scenario which represent real user interactions. All next test cases in that area should use nailgun